<!-- -*- mode: XML -*- -->
<!-- run velodyne_pointcloud/CloudNodelet in a nodelet manager -->

<launch>
  <arg name="calibration" default="" />
  <arg name="manager" default="pandar_nodelet_manager" />
  <arg name="max_range" default="130.0" />
  <arg name="min_range" default="0.9" />
  <arg name="start_angle" default="0" />

  <arg name="device_ip" default="" />
  <arg name="frame_id" default="pandar" />
  <arg name="model" default="" />
  <arg name="pcap" default="" />
  <arg name="port" default="8080" />
  <arg name="read_fast" default="false" />
  <arg name="read_once" default="false" />
  <arg name="repeat_delay" default="0.0" />
  <arg name="rpm" default="600.0" />

  <node pkg="nodelet" type="nodelet" name="$(arg manager)_cloud"
        args="load pandar_pointcloud/CloudNodelet $(arg manager)"
		output="screen">
    <param name="calibration" value="$(arg calibration)"/>
    <param name="max_range" value="$(arg max_range)"/>
    <param name="min_range" value="$(arg min_range)"/>
    <param name="start_angle" value="$(arg start_angle)"/>

    <param name="device_ip" value="$(arg device_ip)" />
    <param name="frame_id" value="$(arg frame_id)"/>
    <param name="model" value="$(arg model)"/>
    <param name="pcap" value="$(arg pcap)"/>
    <param name="port" value="$(arg port)" />
    <param name="read_fast" value="$(arg read_fast)"/>
    <param name="read_once" value="$(arg read_once)"/>
    <param name="repeat_delay" value="$(arg repeat_delay)"/>
    <param name="rpm" value="$(arg rpm)"/>
  </node>

  <!--node pkg="nodelet" type="nodelet" name="$(arg manager)_color"
        args="load pandar_pointcloud/RingColorsNodelet $(arg manager)"
		output="screen"/-->
</launch>
